src/main/groovy/org/transmartproject/batch/highdim/mrna/data/MrnaDataJobConfiguration.groovy

```diff
@@ -59,10 +59,9 @@ class MrnaDataJobConfiguration extends AbstractStandardHighDimJobConfiguration {
                 partitionByColumn: 'trial_name',
                 partitionByColumnValue: studyId,
                 sequence: Sequences.MRNA_PARTITION_ID,
-                indexes: [
-                        ['assay_id'],
-                        ['probeset_id'],
-                        ['patient_id'],
+                primaryKey: ['assay_id', 'probeset_id'],
+                foreignKeys: [
+                        assay_id: [table: Tables.SUBJ_SAMPLE_MAP, column: 'assay_id', onDelete: 'CASCADE']
```

Comment:
After this, `patient_id` and `probeset_id` have no index. For patient_id, that's probably not a problem (though I haven't looked at the access patterns), but for `probeset_id` it likely is since that column is used to join with the annotations table.